### PR TITLE
chore(commands): remove most type errors

### PR DESCRIPTION
**Pull request by branchvincent**
# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -97,6 +97,7 @@ force-exclude = '''
 [tool.mypy]
 check_untyped_defs = true
 ignore_missing_imports = true
+show_error_codes = true
```
Comment (branchvincent) [inline pyproject.toml:100]: this [shows error codes](https://mypy.readthedocs.io/en/stable/error_codes.html#displaying-error-codes), which was useful for our flake8 lint that requires `type: ignore[error-code]`

File: src/poetry/console/commands/export.py [hunk 2]
```diff
@@ -48,11 +48,11 @@ def handle(self) -> None:
             self.line_error("<comment>The lock file does not exist. Locking.</comment>")
             options = []
             if self.io.is_debug():
-                options.append(("-vvv", None))
+                options.append("-vvv")
```
Comment (branchvincent) [inline src/poetry/console/commands/export.py:51]: Found a bug here 😄

```console
$ rm poetry.lock && poetry export -vv
...
  TypeError

  sequence item 0: expected str instance, tuple found
  at src/poetry/console/commands/export.py:57 in handle
       53│                 options.append(("-vv", None))
       54│             elif self.io.is_verbose():
       55│                 options.append(("-v", None))
       56│ 
    →  57│             self.call("lock", " ".join(options))
       58│ 
       59│         if not locker.is_fresh():
       60│             self.line_error(
       61│                 ""
```

File: src/poetry/console/commands/source/add.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def handle(self) -> Optional[int]:
                 )
                 return 1
 
-            if source.name == name:
+            if new_source and source.name == name:
```
Comment (branchvincent) [inline src/poetry/console/commands/source/add.py:89]: Without this check, then the below assignment `source = new_source` maybe have produced `source = None`

File: src/poetry/console/commands/config.py [hunk 4]
```diff
@@ -158,20 +159,19 @@ def handle(self) -> Optional[int]:
...
+                if setting_key not in self.unique_config_values:
                     raise ValueError(f"There is no {setting_key} setting.")
 
                 value = config.get(setting_key)
 
                 if not isinstance(value, str):
-                    value = json.dumps(value)
-
-                self.line(value)
+                    self.line(json.dumps(value))
```
Comment (neersighted) [inline src/poetry/console/commands/config.py:168]: This logic is a bit jumbled, and I feel this change makes it moderately worse. Can we express this idiom better?

File: src/poetry/console/commands/add.py [hunk 5]
```diff
@@ -236,7 +237,7 @@ def handle(self) -> int:
         if self.option("lock"):
             self._installer.lock()
 
-        self._installer.whitelist([r["name"] for r in requirements])
+        self._installer.whitelist([cast(str, r["name"]) for r in requirements])
```
Comment (branchvincent) [inline src/poetry/console/commands/add.py:240]: reached for a `cast` here as `requirements` is too loosely typed as `Dict[str, Union[str, List[str]]]`. planning to tighten that up after removing all ignored files

File: src/poetry/console/commands/plugin/add.py [hunk 6]
```diff
@@ -112,9 +112,11 @@ def handle(self) -> int:
                 root_package = ProjectPackage(package.name, package.version)
                 for dependency in package.requires:
                     root_package.add_dependency(dependency)
-
                 break
 
+        if not root_package:
```
Comment (neersighted) [inline src/poetry/console/commands/plugin/add.py:117]: This change is good, but seems to be unrelated. Can we break it out into a separate PR?

File: src/poetry/console/commands/remove.py [hunk 7]
```diff
@@ -83,8 +83,8 @@ def handle(self) -> int:
         if "group" in poetry_content and not poetry_content["group"]:
             del poetry_content["group"]
 
-        removed = set(removed)
-        not_found = set(packages).difference(removed)
+        removed_ = set(removed)
```
Comment (neersighted) [inline src/poetry/console/commands/remove.py:86]: Not sure I like the underscore suffix here. Why not `removed_set` or something similar?

File: src/poetry/console/commands/shell.py [hunk 8]
```diff
@@ -32,6 +33,9 @@ def handle(self) -> None:
 
         self.line(f"Spawning shell within <info>{self.env.path}</>")
 
+        if not isinstance(self.env, VirtualEnv):
```
Comment (neersighted) [inline src/poetry/console/commands/shell.py:36]: This should also be factored out into its own change.

File: src/poetry/console/commands/version.py [hunk 9]
```diff
@@ -80,27 +80,27 @@ def increment_version(self, version: str, rule: str) -> "Version":
         from poetry.core.semver.version import Version
 
         try:
-            version = Version.parse(version)
+            version_ = Version.parse(version)
```
Comment (neersighted) [inline src/poetry/console/commands/version.py:83]: Same as above, I'd rather call this `parsed` or similar.

**Review by neersighted — CHANGES_REQUESTED** [review]

File: src/poetry/console/commands/add.py (hunk 5 above)
Comment (neersighted) [reply src/poetry/console/commands/add.py:240]: That's kind of what I expected, sounds good.

File: src/poetry/console/commands/version.py (hunk 9 above)
Comment (branchvincent) [reply src/poetry/console/commands/version.py:83]: almost did that originally

File: src/poetry/console/commands/plugin/add.py (hunk 6 above)
Comment (branchvincent) [reply src/poetry/console/commands/plugin/add.py]: yup, just added `type: ignore` comments instead

File: src/poetry/console/commands/config.py (hunk 4 above)
Comment (branchvincent) [reply src/poetry/console/commands/config.py:168]: reverted

**Review by neersighted — APPROVED** [review]
